server/app/middlewares/verifyEmail.js

```diff
@@ -0,0 +1,103 @@
+import Auth from '../helpers/auth';
+import Mailer from '../helpers/nodemailer';
+import models from '../../db/models'
+
+const { User } = models;
+
+export default class VerifyEmail {
+	/**
+    * @description generates email verification link and send to new user email
+    * @param req express request object
+    * @param res express response object
+    * @returns next() middleware
+    */
+	static async generateEmailLink(req, res, next) {
+		const {firstName, lastName, email, password } = req.body;
+		const token = Auth.generateMailToken({firstName, lastName, email, password});
+		const link = `https://localhost:3000/verifyEmail?token=${token}`;
+		const sectionStyle = 'color: #fff; background-color: grey; ';
+		const linkStyle = `padding: 20px; 
+						   border-radius: 4px; 
+						   text-align: center; 
+						   font-weight: bold; 
+						   background-color: #5D45E3`;
+
+		const body = `<section style=${style}>
+						<h2 style='text-align: center; color: #5D45E3 '>Greetings ${firstName}!</h2>
+						<div> 
+							You have successfully registered an account with HelloBooks website.
+							To activate your account, kindly confirm your registration.
+
+							<a href='${link}' style=${linkStyle}> CONFIRM REGISTRATION</a>
+
+							<p>
+								Note: If you do not use this link before 1hour it
+								will expire, and you may need to sign up again.
+								Any questions? send us an email info@hellobooks.com.ng
+							</p>
+						</div>	
+					</section>
+					`;
+
+		const mailObj = {
+			email,
+			subject: 'Verify Email',
+			body,
+		};
+		try {
+			await Mailer.notify(mailObj);
```

Comment:
Why do you need to await this? I don't see the point. The next line as well, you're chaining a `.then` to the findOne method and also awaiting it?